src/Sidebar.jsx

```diff
@@ -0,0 +1,130 @@
+import {React, useState } from "react";
+import HomeIcon from "./Images/Home-icon.png";
+import Search from "./Images/search.png";
+import Detail from "./Images/detail-icon.png";
+import BarOpener from "./Images/bar-opener.png";
+import Profile from "./Images/profile logo.png";
+
+export default function Sidebar() {
+  const [Sidebar, setSidebar] = useState(true);
+  const [CurrentSelectedItem, setCurrentSelectedItem] = useState(null);
+
+  const ToggleSidebar = () => {
+    setSidebar(!Sidebar);
+  };
+  return (
+    <div>
+      <nav
+        className={`bg-gray-300 fixed h-full rounded-lg ${
+          Sidebar ? "w-36 sm:w-20 md:w-36 duration-500" : "w-20 duration-500"
+        }`}
+      >
+        <ul className="mt-2">
+          <button className="" onClick={ToggleSidebar}>
+            <img className="sm:h-5 hover:bg-slate-100" src={BarOpener} />
+          </button>
+
+          <li
+            className={` hover:bg-slate-100 rounded-md ${
+              CurrentSelectedItem === "All Countries"
+                ? "bg-gray-100"
+                : "bg-gray-400"
+            }`}
+            onClick={() => {
+              setCurrentSelectedItem("All Countries");
+            }}
+          >
+            <button className="flex items-center">
+              <img
+                className={`sm:h-4 sm:px-8 ${
+                  Sidebar ? "md:px-0 md:pl-2 " : "md:px-8"
+                }`}
+                src={HomeIcon}
+                alt="home image"
+              />
+
+              <div
+                className={`${
+                  Sidebar ? "block sm:hidden md:block md:pl-2" : "hidden"
+                }`}
+              >
+                All Countries
+              </div>
+            </button>
+          </li>
+
+          <li
+            className={`bg-gray-400  mt-2  hover:bg-slate-100 rounded-md } ${
+              CurrentSelectedItem === "CountryDetails"
```

Comment:
Use constant for selected Items.

Reply:
You mean instead of giving string direct  store them in to any constant and then compare ?